core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java

```diff
@@ -103,26 +103,31 @@ public RecordCursor cursor(
         listCatalogNames(session, metadata, accessControl, catalogFilter).forEach(catalogName -> {
             QualifiedTablePrefix tablePrefix = tablePrefix(catalogName, schemaFilter, tableFilter);
 
-            getMaterializedViews(session, metadata, accessControl, tablePrefix).forEach((tableName, definition) -> {
-                QualifiedObjectName name = new QualifiedObjectName(tablePrefix.getCatalogName(), tableName.getSchemaName(), tableName.getTableName());
-                MaterializedViewFreshness freshness;
-
-                try {
-                    freshness = metadata.getMaterializedViewFreshness(session, name);
-                }
-                catch (MaterializedViewNotFoundException e) {
-                    // Ignore materialized view that was dropped during query execution (race condition)
-                    return;
-                }
-
-                Object[] materializedViewRow = createMaterializedViewRow(name, freshness, definition);
-                displayTable.addRow(materializedViewRow);
-            });
+            addMaterializedViewForCatalog(session, displayTable, tablePrefix);
         });
 
         return displayTable.build().cursor();
     }
 
+    public void addMaterializedViewForCatalog(Session session, InMemoryRecordSet.Builder displayTable, QualifiedTablePrefix tablePrefix)
```

Comment:
Extraction helper methods for clarity makes total sense. But please leave those as `private` as they are not used externally.